crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -1100,6 +1311,9 @@ declare_lint! {
     /// ## What it does
     /// Makes sure that instance attribute accesses are valid.
     ///
+    /// ## Why is this bad?
+    /// Incorrect attribute accesses raise `AttributeError` at runtime.
```

Comment:
hmm, not necessarily. In the example given immediately below, the assignment won't raise `AttributeError`. It's just that the user has explicitly stated that assignments on instances should be prohibited by annotating the variable with `ClassVar`.